From 012e39a060440077f24c6df6b4c9ce77032d175f Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Wed, 20 Nov 2019 10:23:35 -0600
Subject: [PATCH 09/34] ARM64: dts: logicpd-imx8mm: Fix SDHC2 pin conflict

The SDHC2 is currently configured to use GPIO instead of the CD_B pin.
This GPIO causes a conflict and has been marked as 'non-removable' when
in reality it is removable.

This patch changes the GPIO pin configuration to USDHC2_CD_B and removes
the non-removable flag from the device tree.

Signed-off-by: Adam Ford <adam.ford@logicpd.com>
---
 arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi b/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi
index 7c68f653b92c..4b62b45a0f56 100644
--- a/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi
+++ b/arch/arm64/boot/dts/freescale/lpd-imx8mm-baseboard.dtsi
@@ -186,7 +186,7 @@
 
 		pinctrl_usdhc2_gpio: usdhc2grpgpio {
 			fsl,pins = <
-				MX8MM_IOMUXC_GPIO1_IO15_GPIO1_IO15	0x1c4
+				MX8MM_IOMUXC_SD2_CD_B_USDHC2_CD_B	0x41
 				MX8MM_IOMUXC_SD2_RESET_B_GPIO2_IO19	0x41
 			>;
 		};
@@ -392,7 +392,6 @@
 	pinctrl-1 = <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
 	pinctrl-2 = <&pinctrl_usdhc2_200mhz>, <&pinctrl_usdhc2_gpio>;
 	bus-width = <4>;
-	non-removable;
 	vmmc-supply = <&reg_usdhc2_vmmc>;
 	status = "okay";
 };
-- 
2.17.1

